File: backend/src/main/java/com/example/backend/service/IssueService.java
package com.example.backend.service;

import com.example.backend.entity.IssueBook;
import java.util.List;

public interface IssueService {

    IssueBook issueBook(IssueBook issueBook);

    List<IssueBook> getAllIssuedBooks();

    IssueBook getIssueById(Long id);

    IssueBook returnBook(Long id);

    void deleteIssue(Long id);

}

File: frontend/src/services/issueService.js
import api from "./api";

const getAllIssuedBooks = async () => {
  const response = await api.get("/issues");
  return response.data;
};

const issueBook = async (issueData) => {
  const response = await api.post("/issues", issueData);
  return response.data;
};

const returnBook = async (id) => {
  const response = await api.put(`/issues/return/${id}`);
  return response.data;
};

const getIssueById = async (id) => {
  const response = await api.get(`/issues/${id}`);
  return response.data;
};

const issueService = {
  getAllIssuedBooks,
  issueBook,
  returnBook,
  getIssueById,
};

export default issueService;